web/js/codeworld_shared.js

```diff
@@ -532,12 +537,48 @@ function registerStandardHints(successFunc) {
       found[2] = [];
     }
 
-    const options = found[0].concat(found[1]).concat(found[2]);
+    const mappedTerms = {
+      square: 'Prelude.rectangle',
+      triangle: 'Prelude.polygon',
+    };
+
+    for (const [lookedUpTerm, replacementTerm] of Object.entries(mappedTerms)) {
+      let matchingCandidate;
+      for (const bucket of found) {
+        matchingCandidate = bucket.find(({ text }) =>
+          replacementTerm.includes(text)
+        );
+
+        if (matchingCandidate) {
+          break;
+        }
+      }
+
+      if (window.codeWorldSymbols[replacementTerm] && !matchingCandidate) {
```

Comment:
I'm confused about why you're checking `!matchingCandidate`.  In fact, I've pretty much convinced myself that `matchingCandidate` will always be true if the symbol is in `codeWorldSymbols`.  We just finished iterating over every key in `codeWorldSymbols` and adding it to `found`.  Am I wrong?

That's actually an interesting point, though.  All of these completions **already** exist in `found[2]`, so maybe the task here isn't to add a new completion, but to adjust the scores to promote them.  That is, the ultimate score for a choice in bucket 2 should be the maximum score of any target term that redirects to this one (after applying the redirect penalty).  That's a completely different implementation, so... sorry about that.  Worth thinking about, though.